sdk/src/test/java/io/dapr/client/DaprClientGrpcTest.java

```diff
@@ -1927,6 +1928,74 @@ public void getSecretsWithMetadata() {
     assertEquals(expectedValue, result.get(expectedKey));
   }
 
```

Comment:
I had created a test for nullable etag, if you'd like to include it:

```suggestion

  @Test
  public void saveStateTestNullEtag() {
    String key = "key1";
    String etag = null;
    String value = "State value";
    ArgumentCaptor<DaprProtos.SaveStateRequest> argument = ArgumentCaptor.forClass(DaprProtos.SaveStateRequest.class);
    doAnswer((Answer<Void>) invocation -> {
      StreamObserver<Empty> observer = (StreamObserver<Empty>) invocation.getArguments()[1];
      observer.onNext(Empty.getDefaultInstance());
      observer.onCompleted();
      return null;
    }).when(daprStub).saveState(argument.capture(), any());

    StateOptions options = buildStateOptions(StateOptions.Consistency.STRONG, StateOptions.Concurrency.FIRST_WRITE);
    Mono<Void> result = client.saveState(STATE_STORE_NAME, key, etag, value, options);

    result.block();
    assertFalse(argument.getValue().getStates(0).hasEtag());
  }
```

Reply:
Actually, why don't I just create a separate PR with the tests, cause I wrote one for deleteState as well.

Reply:
https://github.com/dapr/java-sdk/pull/454

Reply:
Ack.